elevate-cpanel

```diff
@@ -3035,7 +3039,7 @@ sub is_script_out_of_date ($self) {
         $blocker_text = "The script could not fetch information about the latest version.";
     }
     else {
-        $should_block = $latest_version eq $self_version ? 0 : 1;
+        $should_block = $latest_version == $self_version ? 0 : 1;
```

Comment:
I'm concerned that we're not validating `$latest_version` and `$self_version` are integers before doing an == check.

Reply:
`$self_version` is just the `VERSION` constant, so we can guarantee it's an integer.
`$latest_verson` eventually calls `_latest_version()` as a builder:

```perl
sub _latest_version ($self) {
    my $http = Cpanel::HTTP::Client->new->die_on_http_error();
    my $url  = $self->ELEVATE_BASE_URL;
    my $response = $http->get( $url . '/version' );
    return int $response->content();
}
```

This dies on errors, and it relies on the URL returning something that can be interpreted by `int()`. In principle, a rogue proxy could cause a client to see the empty string.